Kernel/Storage/Ramdisk/Device.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Memory.h>
+#include <AK/StringView.h>
+#include <Kernel/Devices/DeviceManagement.h>
+#include <Kernel/FileSystem/OpenFileDescription.h>
+#include <Kernel/Storage/Ramdisk/Controller.h>
+#include <Kernel/Storage/Ramdisk/Device.h>
+
+namespace Kernel {
+
+NonnullLockRefPtr<RamdiskDevice> RamdiskDevice::create(RamdiskController const& controller, NonnullOwnPtr<Memory::Region>&& region, int major, int minor)
+{
+    auto device_or_error = DeviceManagement::try_create_device<RamdiskDevice>(controller, move(region), major, minor);
+    // FIXME: Find a way to propagate errors
```

Comment:
it seems you can safely return `ErrorOr`? the calling code has no error handling too, so you would have everything in one place! 😉

Reply:
Again, I'm just reverting the old code. I think it is outside the scope of this PR to clean up that code, since it will be removed again once the initramfs code is upstream.